src/main/kotlin/com/github/hanseter/json/editor/ui/FilterableTreeItem.kt

```diff
@@ -105,7 +106,7 @@ interface TreeItemData {
     val required: Boolean
     val cssClasses: List<String>
     val cssStyle: String?
-    var validationMessage: String?
+    var validationMessage: Pair<Severity, String>?
```

Comment:
I was fine with the Pair as long as we only juggled them around in private methods of a class.
I would much rather prefer a dedicated class for the publicly visible surface area.

Reply:
There even already is a dedicated type `ValidationResult` or am I tripping?